AllReadyApp/Web-App/AllReady/wwwroot/js/ko.validation.js

```diff
@@ -17,4 +17,56 @@ ko.observableArray.fn.uniqueValidator = function (property) {
                 return prev;
             }, []).length;
     });
+};
+
+//
+// This is a "poor man's" validation routine that provides specific, limited capabilities (currently only supports required, email, phoneNumber validations).
+// If more a more general & extensive validation capability is needed, consider using Knockout-Validation plugin at https://github.com/Knockout-Contrib/Knockout-Validation
+//
+ko.extenders.validate = function (target, options) {
```

Comment:
I like what you did here, but the syntax for using `validate` is a bit clunky. For instance, it's not very obvious what it means to set `phoneNumber` equal to the empty string. It kind of forces you to go read the implementation of `validate`. Additionally, the maintainability of the `validate` function is somewhat suspect. Do you just keep adding if blocks to it for each kind of validation you want to implement? And each of the if blocks has basically the same format and should probably be refactored into a single function.
```
self.PreferredPhoneNumber.extend({
    validate: {
        required: "'Phone Number' is required",
        phoneNumber: ""
    }
});
```

I think it might work better to use the `withValidator` fn in a way that you could then chain validation functions together something like
```
self.PreferredPhoneNumber
    .isRequired({ displayName: "Phone Number"})
    .validatePhoneNumber()
```

`withValidator` would need to be enhanced to be chainable. This isn't too bad. You just check whether `isValid` is already defined and combine its result with the result of `validatorFn` when you re-define `isValid` as a new computed observable. Notice I also changed the base type to be `subscribable` so you can use it with arrays or single values. This also achieves the same short-circuiting that your validate function does because `validateFn` will not execute unless `isValidAlready` returns true.
```
ko.subscribable.fn.withValidator = function (validatorFn) {
    var isValidAlready = this.isValid || function () { return true; } //Default to valid if no validator defined yet
    this.isValid = ko.computed(function () {
        return isValidAlready() && validatorFn(this);
    }, this);
    return this;
};
```

You could also pretty easily add an option to defer validation until the observable changes from its initial value. That seems basically what you are getting at with `showErrors`. If you want to work together on this, I'd be happy to.